logdna/resource_index_rate_alert.go

```diff
@@ -0,0 +1,253 @@
+package logdna
+
+import (
+  "context"
+  "encoding/json"
+  "log"
+
+  "github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+  "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+  "github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+const indexRateAlertConfigID = "config"
+
+/**
+ * Create index rate alert resource
+ * As API does not allow POST method this method calls PUT
+ * which allows upsert and create a new index rate alert config record is not exist
+ * Only one config per account is allowed
+ */
+func resourceIndexRateAlertCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+  var diags diag.Diagnostics
+  pc := m.(*providerConfig)  
+
+  indexRateAlert := indexRateAlertRequest{}
+
+  if diags = indexRateAlert.CreateRequestBody(d); diags.HasError() {
+    return diags
+  }
+
+  req := newRequestConfig(
+    pc,
+    "PUT",
+    "/v1/config/index-rate",
+    indexRateAlert,
+  )
+
+  body, err := req.MakeRequest()
+
+  log.Printf("[DEBUG] %s %s, payload is: %s", req.method, req.apiURL, body)
+
+  if err != nil {
+    return diag.FromErr(err)
+  }
+
+  createdIndexRateAlert := indexRateAlertResponse{}
+  err = json.Unmarshal(body, &createdIndexRateAlert)
+
+  if err != nil {
+    return diag.FromErr(err)
+  }
+
+  log.Printf("[DEBUG] %s %s SUCCESS. Remote resource updated.", req.method, req.apiURL)
+
+  d.SetId(indexRateAlertConfigID)
+
+  return resourceIndexRateAlertRead(ctx, d, m)
+}
+
+func resourceIndexRateAlertUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+  var diags diag.Diagnostics
+  pc := m.(*providerConfig)
+
+  indexRateAlert := indexRateAlertRequest{}
+
+  if diags = indexRateAlert.CreateRequestBody(d); diags.HasError() {
+    return diags
+  }
+
+  req := newRequestConfig(
+    pc,
+    "PUT",
+    "/v1/config/index-rate",
+    indexRateAlert,
+  )
+
+  body, err := req.MakeRequest()
+  log.Printf("[DEBUG] %s %s, payload is: %s", req.method, req.apiURL, body)
+
+  if err != nil {
+    return diag.FromErr(err)
+  }
+
```

Comment:
nit/question: Because of the PUT, is this not the same as create (aside from some minor logging differences)?  Just wondering if we could reduce code duplication by calling `resourceIndexRateAlertCreate()` here?

It's not a TON of code, so it probably doesn't save much, except if the API ever changes, we wouldn't have to change 2 places in the code.

Reply:
Got rid of `resourceIndexRateAlertCreate`.
`resourceIndexRateAlertUpdate` is used in both cases: `Create` and `Update`